test-case/test-mic-privacy.sh

```diff
@@ -0,0 +1,173 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2025 Intel Corporation. All rights reserved.
+
+##
+## Case Name: test-mic-privacy
+##
+## Preconditions:
+##    HW managed mode (Only for DMIC PCH and SNDW interfaces).
+##    This test case requires physical loopback between playback and capture.
+##    playback <=====> capture
+##    USB relay switch is connected. The usbrelay app is installed.
+##    Instruction: https://github.com/darrylb123/usbrelay
+##
+## Description:
+##    Run alsabat process perform both playback and capture.
+##    Enable MIC privacy.
+##    Run alsabat process perform both playback and capture again.
+##
+## Case step:
+##    1. Specify the pcm IDs for playback and capture
+##    2. Check if usbrelay is installed and connected properly.
+##    3. Run alsabat process perform both playback and capture.
+##    4. Switch relay 1 to enable MIC privacy.
+##    5. Run alsabat process perform both playback and capture.
+##
+## Expect result:
+##    After step 3 the return value is 0.
+##    After step 5 the return value is -1001 (no peak be detected).
+
+# remove the existing alsabat wav files
+rm -f /tmp/mc.wav.*
+
+# shellcheck source=case-lib/lib.sh
+source "$(dirname "${BASH_SOURCE[0]}")"/../case-lib/lib.sh
+
+OPT_NAME['p']='pcm_p'     	OPT_DESC['p']='pcm for playback. Example: hw:0,0'
+OPT_HAS_ARG['p']=1          OPT_VAL['p']='hw:0,0'
+
+OPT_NAME['N']='channel_p'   OPT_DESC['N']='channel number for playback.'
+OPT_HAS_ARG['N']=1          OPT_VAL['N']='2'
+
+OPT_NAME['c']='pcm_c'      	OPT_DESC['c']='pcm for capture. Example: hw:0,1'
+OPT_HAS_ARG['c']=1          OPT_VAL['c']='hw:0,1'
+
+OPT_NAME['C']='channel_c'   OPT_DESC['C']='channel number for capture.'
+OPT_HAS_ARG['C']=1          OPT_VAL['C']='2'
+
+OPT_NAME['s']='sof-logger'  OPT_DESC['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_HAS_ARG['s']=0          OPT_VAL['s']=1
+
+OPT_NAME['r']='relay'       OPT_DESC['r']='name of usbrelay switch, default value is HURTM_1'
+OPT_HAS_ARG['r']=1          OPT_VAL['r']="HURTM_1"
+
+func_opt_parse_option "$@"
+
+pcm_p=${OPT_VAL['p']}
+pcm_c=${OPT_VAL['c']}
+channel_c=${OPT_VAL['C']}
+channel_p=${OPT_VAL['N']}
+relay=${OPT_VAL['r']}
+rate=48000
+
+dlogi "Params: pcm_p=$pcm_p, pcm_c=$pcm_c, channel_c=$channel_c, channel_p=$channel_p, rate=$rate, LOG_ROOT=$LOG_ROOT"
+
+function __upload_wav_files
+{
+    # upload the alsabat wav file
+    for file in /tmp/mc.wav.*; do
+        # alsabat has a bug where it creates an empty record in playback mode
+        if test -s "$file"; then
+            cp "$file" "$LOG_ROOT/"
+        fi
+    done
+}
+
+function check_playback_capture
```

Comment:
```suggestion
check_playback_capture
```
Unnecessary bashism.